webpack/components/extensions/HostDetails/Tabs/RepositorySetsTab/RepositorySetsTab.js

```diff
@@ -0,0 +1,347 @@
+    updateSearchQuery, selectNone, fetchBulkParams, ...selectAll
+  } = useBulkSelect({
+    results,
+    metadata,
+    isSelectable: () => false,
+  });
+
+  const hostDetailsStatus = useSelector(state => selectHostDetailsStatus(state));
+
+  useEffect(() => {
+    // wait until host details are loaded to set alertShowing
+    if (hostDetailsStatus === STATUS.RESOLVED) {
+      setAlertShowing(nonLibraryHost);
+    }
+  }, [hostDetailsStatus, nonLibraryHost]);
+
+  if (!hostId) return <Skeleton />;
+
+  const updateResults = ({ labels, enabled, newResponse }) => dispatch({
+    type: `${REPOSITORY_SETS_KEY}_SUCCESS`,
+    key: REPOSITORY_SETS_KEY,
+    response: {
+      ...response,
+      results: results.map((result) => {
+        if (labels.includes(result.label)) {
+          const isEnabled = enabled === null ?
+            newResponse.results.find(r => r.id === result.id).enabled :
+            enabled;
+          return { ...result, enabled: isEnabled, enabled_content_override: enabled };
+        }
+        return result;
+      }),
+    },
+  });
+  const overrideToEnabled = ({ labels }) =>
+    dispatch(enableRepoSetRepo({ hostId, labels, updateResults }));
+  const overrideToDisabled = ({ labels }) =>
+    dispatch(disableRepoSetRepo({ hostId, labels, updateResults }));
+  const resetToDefault = ({ labels }) =>
+    dispatch(resetRepoSetRepo({ hostId, labels, updateResults }));
+
+  // uncomment (and remove rule disablement) when Select All is added
+  /* eslint-disable max-len */
+  // const dropdownItems = [
+  //   <DropdownItem aria-label="bulk_enable" key="bulk_enable" component="button" onClick={overrideToEnabled}>
+  //     {__('Override to enabled')}
+  //   </DropdownItem>,
+  //   <DropdownItem aria-label="bulk_disable" key="bulk_disable" component="button" onClick={overrideToDisabled}>
+  //     {__('Override to disabled')}
+  //   </DropdownItem>,
+  //   <DropdownItem aria-label="bulk_disable" key="bulk_disable" component="button" onClick={resetToDefault}>
+  //     {__('Reset to default')}
+  //   </DropdownItem>,
+  // ];
+  /* eslint-enable max-len */
+
+  let toggleGroup;
+  if (nonLibraryHost) {
+    toggleGroup = (
+      <ToggleGroup aria-label="Repository Set toggle">
+        <ToggleGroupItem
+          text={__('Show all')}
+          buttonId="no-limit-toggle"
+          aria-label="No limit"
+          isSelected={toggleGroupState === noLimit}
+          onChange={() => setToggleGroupState(noLimit)}
+        />
+        <ToggleGroupItem
+          text={__('Limit to environment')}
+          buttonId="limit-to-env-toggle"
+          aria-label="Limit to environment"
+          isSelected={toggleGroupState === limitToEnvironment}
+          onChange={() => setToggleGroupState(limitToEnvironment)}
+        />
+      </ToggleGroup>
+    );
+  }
+
+  const hostEnvText = 'the "{contentViewName}" content view and "{lifecycleEnvironmentName}" environment';
```

Comment:
Is there a better way to do this, Feels liek we'd lose out on i18n

Reply:
@parthaa Check out the documentation for FormattedMessage from react-intl.  I think what I have will still be translated correctly.

Reply:
lgtm ack